app/src/main/java/com/example/stromtracker/ui/urlaub/UrlaubFragment.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.stromtracker.ui.urlaub
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.stromtracker.R
+import com.example.stromtracker.database.Geraete
+import com.example.stromtracker.database.Urlaub
+import com.example.stromtracker.ui.SharedViewModel
+import com.example.stromtracker.ui.urlaub.urlaub_new.UrlaubNewFragment
+import com.getbase.floatingactionbutton.FloatingActionButton
+
+class UrlaubFragment : Fragment(), View.OnClickListener {
+
+    private lateinit var urlaubViewModel: UrlaubViewModel
+    private lateinit var sharedViewModel: SharedViewModel
+
+    private lateinit var urlaubList: ArrayList<Urlaub>
+    private lateinit var verbraucherList: ArrayList<Geraete>
+
+    private lateinit var recyclerView: RecyclerView
+    private lateinit var viewAdapter: RecyclerView.Adapter<*>
+    private lateinit var viewManager: RecyclerView.LayoutManager
+
+    private lateinit var addUrlaub: FloatingActionButton
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        urlaubViewModel = ViewModelProvider(this).get(UrlaubViewModel::class.java)
+        sharedViewModel = ViewModelProvider(requireActivity()).get(SharedViewModel::class.java)
+
+        urlaubList = ArrayList()
+        verbraucherList = ArrayList()
+
+        val root = inflater.inflate(R.layout.fragment_urlaub, container, false)
+
+        //RecyclerView initialisieren
+        viewAdapter = UrlaubListAdapter(urlaubList, sharedViewModel.getHaushalt().value!!)
+
+        viewManager = LinearLayoutManager(this.context)
+        recyclerView = root.findViewById<RecyclerView>(R.id.urlaub_recycler_view).apply {
+            setHasFixedSize(true)
+            layoutManager = viewManager
+            adapter = viewAdapter
+        }
```

Comment:
Warum genau machst du das?

Reply:
Um die recyclerview zu instanziieren?

Habe ich bei Geräte genau so gemacht, machst du es anders oder wie?

Reply:
Ich hab da nichts mit set setfixed size

Reply:
Hm, ist anscheinend für Optimierungen da.
https://stackoverflow.com/questions/28709220/understanding-recyclerview-sethasfixedsize
https://stackoverflow.com/questions/28827597/when-do-we-use-the-recyclerview-sethasfixedsize/40707099

Soweit ich verstehe, soll das true sein wenn sich die Größe der Recyclerview (nicht der Items) nicht aufgrund des Contents ändert.

Hast du bei dir die reyclerview auf wrap_content?

Reply:
Ah ne du hast es auf match_parent, interessant

Reply:
Ich glaube tatsächlich bei match_parent sollte es auf true sein, weil da die Größe bereits fest ist, maximal so groß wie der parent.

Bei wrap_content dagegen auf false, da es sich dann vergrößert je mehr Items dazukommen.

Schaut aber bitte ihr nochmal nach, nicht dass ich das falsch verstehe

Reply:
Wobei er das bei seinem Blatt nicht setzt, (und default ist false) obwohl er match_parent macht